clang/lib/Driver/Driver.cpp

```diff
@@ -4908,11 +4960,25 @@ class OffloadingActionBuilder final {
         }
       }
 
-      DeviceLinkerInputs.resize(ToolChains.size());
-      bool GpuInitHasErrors = initializeGpuArchMap();
-      if (ShouldAddDefaultTriple)
-        addSYCLDefaultTriple(C, SYCLTripleList);
-      return GpuInitHasErrors;
+      if (ShouldAddDefaultTriple) {
+        if (addSYCLDefaultTriple(C, SYCLTripleList)) {
```

Comment:
The defined order of condition evaluation should allow to reduce the nesting:
```suggestion
      if (ShouldAddDefaultTriple && addSYCLDefaultTriple(C, SYCLTripleList)) {
```